wdl/TasksClusterBatch.wdl

```diff
@@ -0,0 +1,504 @@
+                                      mem_gb: 3.75,
+                                      disk_gb: ceil(base_disk_gb + input_size * 3.0),
+                                      cpu_cores: 1,
+                                      preemptible_tries: 3,
+                                      max_retries: 1,
+                                      boot_disk_gb: 10
+                                  }
+    RuntimeAttr runtime_override = select_first([runtime_attr_override, runtime_default])
+    runtime {
+        memory: "~{select_first([runtime_override.mem_gb, runtime_default.mem_gb])} GB"
+        disks: "local-disk ~{select_first([runtime_override.disk_gb, runtime_default.disk_gb])} HDD"
+        cpu: select_first([runtime_override.cpu_cores, runtime_default.cpu_cores])
+        preemptible: select_first([runtime_override.preemptible_tries, runtime_default.preemptible_tries])
+        maxRetries: select_first([runtime_override.max_retries, runtime_default.max_retries])
+        docker: sv_base_mini_docker
+        bootDiskSizeGb: select_first([runtime_override.boot_disk_gb, runtime_default.boot_disk_gb])
+    }
+
+    command <<<
+        set -euo pipefail
+        mkdir files
+        # Avoids argument limits encountered with ls; uses hard links avoid issues with local file systems
+        # Note that all filenames must be unique
+        cat ~{write_lines(files)} | xargs -I{} -n1 ln {} files/
+        tar czf ~{prefix}.tar.gz -h -C files/ .
+    >>>
+    output {
+        File out = "~{prefix}.tar.gz"
+    }
+}
+
+task TarFilesFromList {
+    input {
+        File file_list
+        String prefix
+        String sv_base_docker
+        RuntimeAttr? runtime_attr_override
+    }
+
+    RuntimeAttr runtime_default = object {
+                                      mem_gb: 3.75,
+                                      disk_gb: 100,
+                                      cpu_cores: 1,
+                                      preemptible_tries: 3,
+                                      max_retries: 1,
+                                      boot_disk_gb: 10
+                                  }
+    RuntimeAttr runtime_override = select_first([runtime_attr_override, runtime_default])
+    runtime {
+        memory: "~{select_first([runtime_override.mem_gb, runtime_default.mem_gb])} GB"
+        disks: "local-disk ~{select_first([runtime_override.disk_gb, runtime_default.disk_gb])} HDD"
+        cpu: select_first([runtime_override.cpu_cores, runtime_default.cpu_cores])
+        preemptible: select_first([runtime_override.preemptible_tries, runtime_default.preemptible_tries])
+        maxRetries: select_first([runtime_override.max_retries, runtime_default.max_retries])
+        docker: sv_base_docker
+        bootDiskSizeGb: select_first([runtime_override.boot_disk_gb, runtime_default.boot_disk_gb])
+    }
+
+    command <<<
+        set -euo pipefail
+        mkdir files
+        # Note that all filenames must be unique
```

Comment:
Not super necessary but maybe it'd be worth checking (ie run them through `basename` and then `uniq -d`) and failing the workflow if this isn't true? Seems like it might cause a hard-to-track down situation later.

Reply:
Done